packages/app/src/cli/models/extensions/specifications/payments_app_extension_schemas/credit_card_payments_app_extension_schema.ts

```diff
@@ -39,6 +43,62 @@ export const CreditCardPaymentsAppExtensionSchema = BasePaymentsAppExtensionSche
     message: 'supports_installments and supports_deferred_payments must be the same',
   })
 
+export interface CreditCardPaymentsAppExtensionDeployConfigType extends BasePaymentsAppExtensionDeployConfigType {
+  ui_extension_registration_uuid?: string
+
+  // Following are overwritten as they are required for credit card extensions
+  start_refund_session_url: string
+  start_capture_session_url: string
+  start_void_session_url: string
+
+  // MultipleCaptureSchema
+  multiple_capture?: boolean
+
+  // DeferredPaymentsSchema
+  supports_deferred_payments: boolean
+  supports_installments: boolean
+
+  // ConfirmationSchema
+  confirmation_callback_url?: string
+  supports_3ds: boolean
+
+  // CreditCard-specific fields
+  start_verification_session_url?: string
+  ui_extension_handle?: string
+  encryption_certificate: string
+  checkout_payment_method_fields?: {
+    type: 'string' | 'number' | 'boolean'
+    required: boolean
+    key: string
+  }[]
+}
+
+export async function creditCardDeployConfigToCLIConfig(
+  config: CreditCardPaymentsAppExtensionDeployConfigType,
+): Promise<Omit<CreditCardPaymentsAppExtensionConfigType, 'name' | 'type' | 'metafields' | 'targeting'> | undefined> {
+  return {
+    api_version: config.api_version,
+    payment_session_url: config.start_payment_session_url,
+    refund_session_url: config.start_refund_session_url,
+    capture_session_url: config.start_capture_session_url,
+    void_session_url: config.start_void_session_url,
```

Comment:
do we need this whole transform function just to rename these? would it be possible to unify the names here?